TruckMuncher/MapViewController.swift

```diff
@@ -337,6 +350,9 @@ class MapViewController: UIViewController,
                 self.navigationItem.titleView?.alpha = self.showingMenu ? 0.0 : 1.0
                 self.navigationItem.leftBarButtonItem?.tintColor = color.colorWithAlphaComponent(self.showingMenu ? 0.0 : 1.0)
                 self.navigationItem.rightBarButtonItem?.tintColor = color.colorWithAlphaComponent(self.showingMenu ? 0.0 : 1.0)
+            }, completion: { (completed) -> Void in
+                currentView.updateViewWithColor(self.showingMenu ? primaryColor : carouselBackground)
```

Comment:
if you move this line into the animation block within the "if recognizer.state == .Ended {" statement, the background color will finish fading in/out like we want. However, you'll see that it leaves one other problem, which is that the table section headers still behave odd.